core/src/loader.rs

```diff
@@ -1492,7 +1493,123 @@ impl<'gc> Loader<'gc> {
                             &mut activation.context,
                             "complete",
                         );
-                        Avm2::dispatch_event(uc, complete_evt, sound_object);
+                        Avm2::dispatch_event(&mut activation.context, complete_evt, sound_object);
+
+                        let mut call_id3_event = false;
+                        let id3 = activation
+                            .avm2()
+                            .classes()
+                            .id3info
+                            .construct(&mut activation, &[])
+                            .expect("failed to construct ID3Info object");
+                        if let Avm2Object::ScriptObject(id3) = id3 {
```

Comment:
I don't think this `if` is necessary at all. We should be able to use the original `id3` just fine.

Reply:
I remember that this is because I'm not sure whether non-public function can be called from another module or not, so I unwrap it so that I can use public function instead.